include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -596,8 +597,13 @@ class operational_domain_impl
      */
     [[nodiscard]] step_point to_step_point(const operational_domain::parameter_point& pp) const noexcept
     {
-        return {static_cast<std::size_t>((pp.x - params.x_min) / params.x_step),
-                static_cast<std::size_t>((pp.y - params.y_min) / params.y_step)};
+        auto it_x = std::find(x_values.cbegin(), x_values.cend(), pp.x);
+        auto it_y = std::find(y_values.cbegin(), y_values.cend(), pp.y);
+
```

Comment:
What if `it_x` or `it_y` point to the end of the range, i.e., no element was found? Maybe add an `assert` here at least?